packages/nextjs/hooks/scaffold-eth/useScaffoldEventHistory.ts

```diff
@@ -36,30 +40,37 @@ export const useScaffoldEventHistory = <
   blockData,
   transactionData,
   receiptData,
-}: UseScaffoldEventHistoryConfig<TContractName, TEventName, TBlockData, TTransactionData, TReceiptData>) => {
+  watch,
+}: UseScaffoldEventHistoryConfig<TContractName, TEventName, TBlockData, TTransactionData, TReceiptData, TWatch>) => {
   const [events, setEvents] = useState<any[]>();
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState<string>();
+  const [fromBlockUpdated, setFromBlockUpdated] = useState<bigint>(fromBlock);
```

Comment:
as I understand with your updates changing `fromBlock` parameter from outside of hook will not affect results.

So if I wanted let's say get events history from block `1000` first, and changed it to `2000`, hook gave me only events from block `2000`. But now it's just default state, so it still will show events from block `1000`. Or am I missing something?

Reply:
Great question! I didn't test it. fromBlock is on the useEffect hook, so readEvents will be called and then I don't know what will happen :-)

I will take a look at this! Thanks!

Reply:
Yes, if you change fromBlock when watching you will still get the events from the first fromBlock value. It doesn't make sense to change the fromBlock value when using the watch feature, maybe we can add something about it to the documentation.

Reply:
🤔  looks like strange behavior. I think on `fromBlock` change it should reload all events (or missing ones). Let's see what others think

Reply:
Please take a look at the proposed change. Now only the events are appended when polling. 

I think it works as the expected behavior now. Thank you for noticing that!

Reply:
Thanks! I updated it a little bit https://github.com/scaffold-eth/scaffold-eth-2/pull/600, since with `append` param it didn't change `fromBlock` behavior. I'm not sure about hooks deps but fast testing worked fine. Could you take a look?

Reply:
@rin-st Thanks!! Merged!